internal/storage/v2/memory/tenant.go

```diff
@@ -110,6 +110,19 @@ func (t *Tenant) findTraceAndIds(query tracestore.TraceQueryParams) ([]traceAndI
 	return traceAndIds, nil
 }
 
+func (t *Tenant) getTraces(traceIds ...tracestore.GetTraceParams) []ptrace.Traces {
+	t.RLock()
+	defer t.RUnlock()
+	traces := make([]ptrace.Traces, 0)
+	for i := range traceIds {
+		index, ok := t.ids[traceIds[i].TraceID]
+		if ok {
+			traces = append(traces, t.traces[index].trace)
```

Comment:
Here we have a common bug with find functions: we need to clone the traces before returning, because query service can run adjustments on them.